-1

Tested at: 8e0ccbdcc485620e5ea44473289cd683666498c0


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/c8094c463f12a2452bea61660239b80aecd54917
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20241/22733/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20241/22733/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20241/22733/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>Selected class -> pat::DiObjectProxy for ROOT: pat::DiObjectProxy
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/BoostedJetMerger.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/MinPatMETProducer.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/SmearedJetProducer.cc:1:0:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/interface/SmearedJetProducerT.h: In member function 'virtual void SmearedJetProducerT<T>::produce(edm::Event&, const edm::EventSetup&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/interface/SmearedJetProducerT.h:204:52: error: 'm_nomVar' was not declared in this scope
                     std::uint32_t seed = jet0eta + m_nomVar + (lumiNum_uint<<10) + (runNum_uint<<20) + evNum_uint;
                                                    ^
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/modules.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/PATPFJetMETcorrInputProducer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/JetSubstructurePacker.cc 
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/ShiftedPFCandidateProducer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/ShiftedPATObjectProducer.cc 
Selected class -> pat::DiObjectProxy for ROOT: pat::DiObjectProxy
>> Compiling LCG dictionary: tmp/slc6_amd64_gcc530/src/PhysicsTools/PatUtils/src/PhysicsToolsPatUtils/a/PhysicsToolsPatUtils_xr.cc
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/plugins/SmearedJetProducer.cc:1:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2017-09-05-2300/src/PhysicsTools/PatUtils/interface/SmearedJetProducerT.h:204:52: error: use of undeclared identifier 'm_nomVar'
                    std::uint32_t seed = jet0eta + m_nomVar + (lumiNum_uint<<10) + (runNum_uint<<20) + evNum_uint;
                                                   ^
1 error generated.
gmake: *** [tmp/slc6_amd64_gcc530/src/PhysicsTools/PatUtils/plugins/PhysicsToolsPatUtils_plugins/SmearedJetProducer.o] Error 1
>> Compile sequence completed for CMSSW CMSSW_8_0_X_2017-09-05-2300
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/c8094c463f12a2452bea61660239b80aecd54917
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20241/22733/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20241/22733/git-merge-result

--- a/PhysicsTools/PatUtils/interface/SmearedJetProducerT.h
+++ b/PhysicsTools/PatUtils/interface/SmearedJetProducerT.h
@@ -200,8 +200,8 @@ class SmearedJetProducerT : public edm::stream::EDProducer<> {
                     unsigned int runNum_uint = static_cast <unsigned int> (event.id().run());
                     unsigned int lumiNum_uint = static_cast <unsigned int> (event.id().luminosityBlock());
                     unsigned int evNum_uint = static_cast <unsigned int> (event.id().event());
-                    unsigned int jet0pt = uint32_t(jets.empty() ? 0 : jets[0].pt()/0.01);
-                    std::uint32_t seed = jet0pt + (lumiNum_uint<<10) + (runNum_uint<<20) + evNum_uint;
+                    unsigned int jet0eta = uint32_t(jets.empty() ? 0 : jets[0].eta()/0.01);
+                    std::uint32_t seed = jet0eta + (lumiNum_uint<<10) + (runNum_uint<<20) + evNum_uint;
                     m_random_generator.seed(seed);
                 }
             }

